modules/btcpay/src/Server/Factory.php

```diff
@@ -21,147 +21,132 @@ class Factory
+			\PrestaShopLogger::addLog(sprintf('[ERROR] %s', $e->getMessage()), $e->getCode());
 			throw new BTCPayException($e->getMessage(), $e->getCode(), $e);
 		}
 
 		// If another BTCPay invoice was created before, returns the original one
 		if (null !== ($redirect = $client->getBTCPayRedirect($cart))) {
-			\PrestaShopLogger::addLog(
-				'[WARNING] Existing BTCPay invoice has already been created, redirecting to it...',
-				2
-			);
+			\PrestaShopLogger::addLog('[WARNING] Existing BTCPay invoice has already been created, redirecting to it...', 2);
 
 			return $redirect;
 		}
 
-		// Setup the Invoice
-		$invoice = new Invoice();
-		$invoice->setFullNotifications(true);
-		$invoice->setExtendedNotifications(true);
-
-		// Get and set transaction speed
-		if (empty($transactionSpeed = \Configuration::get('BTCPAY_TXSPEED'))) {
-			$transactionSpeed = 'default';
-		}
-
-		$invoice->setTransactionSpeed($transactionSpeed);
-
-		// Set an order reference instead of ID, since we haven't made one yet
-		$invoiceReference = \Tools::passwdGen(20);
-		$invoice->setOrderId($invoiceReference);
-
-		// Get shopping currency, currently tested with be EUR
-		$currency       = \Currency::getCurrencyInstance((int) $cart->id_currency);
-		$btcpayCurrency = new BTCPayCurrency($currency->iso_code);
-		$invoice->setCurrency($btcpayCurrency);
-
-		// Add a basic item to our invoice
-		$orderTotal = $cart->getOrderTotal(true);
-		$invoice->setItem(
-			(new Item())
-				->setCode('Cart: ' . $cart->id)
-				->setDescription('Your purchase')
-				->setPrice($orderTotal)
-		);
-
-		// Set POS data so we can verify later on that the call is legit
-		$invoice->setPosData($customer->secure_key);
-
-		// Create a buyer we can add to the invoice
-		$buyer = new Buyer();
-
-		// Add customer information to buyer
-		$customer = new \Customer((int) $cart->id_customer);
-		$buyer->setEmail($customer->email);
-		$buyer->setFirstName($customer->firstname);
-		$buyer->setLastName($customer->lastname);
-
-		// Add address information to buyer
-		$address = new \Address($cart->id_address_delivery);
-		$buyer->setAddress([$address->address1, $address->address2]);
-		$buyer->setCountry($address->country);
-		$buyer->setZip($address->postcode);
-		$buyer->setCity($address->city);
-
-		// Add the state if available
-		if (0 !== ($stateId = $address->id_state)) {
-			$buyer->setState((new \State($stateId))->name);
-		}
-
-		// Finally, add the build buyer to the invoice
-		$invoice->setBuyer($buyer);
+		try {
+			// Setup some stuff
+			$currency         = \Currency::getCurrencyInstance($cart->id_currency);
+			$address          = new \Address($cart->id_address_delivery);
+			$invoiceReference = \Tools::passwdGen(20);
+
+			// Get totals
+			$taxAmount  = $cart->getOrderTotal(true, \Cart::ONLY_PRODUCTS) - $cart->getOrderTotal(false, \Cart::ONLY_PRODUCTS);
+			$orderTotal = (string) $cart->getOrderTotal(true);
+
+			// Build metadata
```

Comment:
You maybe want to make this data optional for privacy reasons. Old api had this but not sure if it is really needed to to have customer data on BTCPay too. For Drupal and now for WP I have it as an option which is off by default. But ymmv